app/misc/git.ts

```diff
@@ -125,6 +125,9 @@ function stage() {
 
 function addAndCommit() {
   commitMessage = document.getElementById('commit-message-input').value;
+  if (commitMessage == null || commitMessage == "") {
+    commitMessage = document.getElementById('commit-message-input-navbar').value;
+  }
   if (commitMessage == null || commitMessage == "") {
     displayModal("Cannot commit without a commit message. Please add a commit message before committing");
     return;
```

Comment:
This sequence of if blocks would benefit from some comments explaining which commit message is used for the git commit and why.

Reply:
Good idea. I have added some comments to explain the sequence of if statements and getting value of elements from 2 different ids